tests/integration/test_basic.py

```diff
@@ -234,3 +235,50 @@ def test_corp_response_header(mini_sentry, relay, cross_origin_resource_policy):
     assert (
         response.headers["cross-origin-resource-policy"] == cross_origin_resource_policy
     )
+
+
+def send_transaction_with_dsc(mini_sentry, relay, project_id, sampling_project_key):
+    relay = relay(mini_sentry)
+
+    now = datetime.datetime.now(datetime.UTC)
+    start_timestamp = (now - datetime.timedelta(minutes=1)).timestamp()
+    timestamp = now.timestamp()
+
+    relay.send_transaction(
+        project_id,
+        payload={
+            "type": "transaction",
+            "transaction": "foo",
+            "start_timestamp": start_timestamp,
+            "timestamp": timestamp,
+            "contexts": {
+                "trace": {
+                    "trace_id": "1234F60C11214EB38604F4AE0781BFB2",
+                    "span_id": "ABCDFDEAD5F74052",
+                    "type": "trace",
+                }
+            },
+        },
+        trace_info={
+            "public_key": sampling_project_key,
+            "trace_id": "1234F60C11214EB38604F4AE0781BFB2",
+        },
+    )
+
+    return mini_sentry.captured_events.get(timeout=1).get_transaction_event()
+
+
+def test_root_project_disabled(mini_sentry, relay):
+    project_id = 42
+    mini_sentry.add_full_project_config(project_id)
+    disabled_dsn = "00000000000000000000000000000000"
+    txn = send_transaction_with_dsc(mini_sentry, relay, project_id, disabled_dsn)
+    assert txn
```

Comment:
This test fails with the previous implementation in https://github.com/getsentry/relay/pull/3770.